Tests/MongoSwiftSyncTests/SpecTestRunner/SpecTest.swift

```diff
@@ -259,35 +279,93 @@ extension SpecTest {
+            _ = try? client.db(dbName).createCollection(collName)
+            // Run the distinct command before every test to prevent `StableDbVersion` error in sharded cluster
+            // transactions. This workaround can be removed once SERVER-39704 is resolved.
+            _ = try? client.db(dbName).collection(collName).distinct(fieldName: "_id")
+        }
+
         if let failPoint = self.failPoint {
             try parent.activateFailPoint(failPoint)
         }
         defer { parent.disableActiveFailPoint() }
 
-        let client = try MongoClient.makeTestClient(options: clientOptions)
-        let monitor = client.addCommandMonitor()
-
-        let db = client.db(dbName)
-        var collection: MongoCollection<Document>?
+        // The spec tests refer to the sessions below. Proactively start them in case this spec test requires one.
+        let sessionsToStart = ["session0", "session1"]
 
-        if let collName = collName {
-            collection = db.collection(collName)
+        var sessionDict = [String: ClientSession]()
+        for session in sessionsToStart {
+            sessionDict[session] = client.startSession(options: self.sessionOptions?[session])
         }
 
+        var sessionIds = [String: Document]()
+
         try monitor.captureEvents {
             for operation in self.operations {
                 try operation.validateExecution(
                     client: client,
-                    database: db,
-                    collection: collection,
-                    session: nil
+                    dbName: dbName,
+                    collName: collName,
+                    sessionDict: sessionDict
                 )
             }
+            // Keep track of the session IDs assigned to each session.
+            // Deinitialize each session thereby implicitly ending them.
+            for session in sessionDict.keys {
+                sessionIds[session] = sessionDict[session]?.id
+                sessionDict[session] = nil
+            }
+        }
+
+        let events = try monitor.commandStartedEvents().map { commandStartedEvent in
+            try processCommandStartedEvent(commandStartedEvent: commandStartedEvent, sessionIds: sessionIds)
         }
-        let events = monitor.commandStartedEvents().map { TestCommandStartedEvent(from: $0) }
 
         if let expectations = self.expectations {
             expect(events).to(match(expectations), description: self.description)
         }
+
+        if let outcome = self.outcome {
+            try self.checkOutcome(outcome: outcome, dbName: dbName, collName: collName!)
+        }
+    }
+
+    internal func processCommandStartedEvent(
+        commandStartedEvent: CommandStartedEvent,
+        sessionIds: [String: Document]
+    )
+        throws -> TestCommandStartedEvent {
+        var event = TestCommandStartedEvent(from: commandStartedEvent)
+        try event.excludeKeys(keys: ["$db", "$clusterTime"])
```

Comment:
this exclusion shouldn't be necessary if we're checking these events using their [`Matchable`](https://github.com/mongodb/mongo-swift-driver/blob/master/Tests/MongoSwiftSyncTests/SpecTestRunner/Match.swift) conformance.

Reply:
Done.